src/ekat/logging/ekat_log_file.hpp

```diff
@@ -0,0 +1,74 @@
+#ifndef EKAT_LOG_FILE_HPP
+#define EKAT_LOG_FILE_HPP
+
+#include "spdlog/spdlog.h"
+#include "spdlog/sinks/null_sink.h"
+#include "spdlog/sinks/basic_file_sink.h"
+#include "spdlog/sinks/rotating_file_sink.h"
+#include <string>
+
+// This file contains policy classes for choosing the behavior of the ekat
+// logger with respect to file output.
+// Each policy must implement the
+//
+// static spdlog::sink_ptr get_file_sink(std::string logfilename)
+//
+// method.
+
+namespace ekat {
+namespace logger {
+
+#define EKAT_LOG_N_FILES 5
+#define EKAT_LOG_MAX_FILE_SIZE_MB 8
+
+// No file output; only console logging.
+struct LogNoFile {
+  static constexpr bool has_filename = false;
+  static std::shared_ptr<spdlog::sinks::null_sink_mt> get_file_sink(const std::string& logfilename="") {
+  return std::make_shared<spdlog::sinks::null_sink_mt>();}
+};
+
+// Basic file output.  Each Logger (usually, 1 per MPI rank) writes to its own
+// file.  No file size limit.
+template <spdlog::level::level_enum FileLogLevel = spdlog::level::debug>
+struct LogBasicFile {
+  static constexpr bool has_filename = true;
+  static std::shared_ptr<spdlog::sinks::basic_file_sink_mt> get_file_sink(
+    const std::string& logfilename) {
+    auto result = std::make_shared<spdlog::sinks::basic_file_sink_mt>(logfilename, true);
+    result->set_level(FileLogLevel);
+    return result;
+  }
+};
+
+
+struct LogSharedFile {
```

Comment:
Can you add a comment? It's intuitively obvious that this logger shares a file with others, but it's not obvious how it works, so one might not be sure what to make of this at first glance.